### ISSUE TITLE
Option on `reduxForm` to automatically throw submission error onSubmit

### ISSUE THREAD

**Issue by wlingke** [erikras/redux-form#3602]
### Are you submitting a **bug report** or a **feature request**?
Feature request

### What is the current behavior?
A frequent use case is calling onSubmit with an ajax call that returns a promise. Currently, if I want that server-side error to show a form-wide error, I have to write something like this in the `catch` block:

```
throw new SubmissionError(error.message)
```

It would be awesome if there was an option on `reduxForm` to automatically throw a submission error from a Promise rejection.

This is kind of similar to how if you return a promise to onSubmit, `submitting` is set to true until the Promise resolves or rejects. It's awesome and super convenient because its a really common use case.

**Comment by gustavohenke** [comment]
Thanks for the suggestion!

**Comment by drew-r** [comment]
Yes please! We have to `.catch(catchAndRethrowAsReduxFormError)` to make our promises play nice.
